subprojects/core-api/src/main/java/org/gradle/api/artifacts/dsl/LockMode.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.artifacts.dsl;
+
+import org.gradle.api.Incubating;
+
+/**
+ * The supported lock modes:
+ * <ul>
+ *     <li>DEFAULT will load the lock state and verify resolution matches it</li>
+ *     <li>STRICT will fail resolution if a locked configuration does not have a lock state</li>
+ *     <li>LENIENT will load the lock state but not perform verification after resolution</li>
```

Comment:
:x: The first reaction I have reading this, is, then why do you load the lock state? It feels like this section deserves a bit more explanation on the use cases. In particular, explain _what_ validation is done and not done.

Reply:
Right, the description of this PR has a more complete explanation. Will improve the javadoc here as well.